src/commands/deploy.ts

```diff
@@ -86,3 +81,18 @@ async function verifyBetaRuntime(outputChannel: vscode.OutputChannel, client: We
         }
     }
 }
+
+async function getFunctionAppNameInPom(pomLocation: string): Promise<string | undefined> {
+    let functionAppName: string | undefined;
+    // tslint:disable-next-line:no-any
+    xml2js.parseString(await fse.readFile(pomLocation, 'utf-8'), { explicitArray: false }, (err: any, result: any): void => {
```

Comment:
Convert this callback into a promise

You can take a look [here](https://github.com/Microsoft/vscode-azurefunctions/blob/master/src/utils/cpUtils.ts#L12) as an example, but here's some pseudo-code:
```
return await new Promise((resolve: () => void, reject: (e: Error) => void): void => {
    xml2js.parseString(pomXmlContents, { explicitArray: false }, (err: any, result: any): void => {
        if (err) {
            reject(err);
        } else {
            resolve(<functionAppName>);
        }
    });
});
```

It has several benefits, but mainly it prevents race-conditions (sometimes a function like `xml2js.parseString` will return before the callback finishes executing)